server/src/main/java/org/elasticsearch/index/engine/InternalEngine.java

```diff
@@ -290,6 +290,7 @@ public InternalEngine(EngineConfig engineConfig) {
             this.internalReaderManager = internalReaderManager;
             this.externalReaderManager = externalReaderManager;
             internalReaderManager.addListener(versionMap);
+            this.lastUnsafeSegmentGenerationForGets.set(lastCommittedSegmentInfos.getGeneration());
```

Comment:
I wonder if we should init the AtomicLong here, i.e.:

```suggestion
            this.lastUnsafeSegmentGenerationForGets = new AtomicLong(lastCommittedSegmentInfos.getGeneration());
```

and remove the initialization on the field? I think it is not used until the engine is constructed.

Reply:
Yeah you are right. It should be safe as you suggested. Pushed 4aab64f7bd493aac6a39419cbec0bdae06c39965